src/main/java/org/jabref/gui/preferences/TableColumnsTabViewModel.java

```diff
@@ -0,0 +1,318 @@
+        );
+
+        if (columnPreferences.getAutoSyncSpecialFieldsToKeyWords() != newColumnPreferences.getAutoSyncSpecialFieldsToKeyWords()) {
+            restartWarnings.add(Localization.lang("Synchronize special fields to keywords"));
+        }
+
+        if (columnPreferences.getSerializeSpecialFields() != newColumnPreferences.getSerializeSpecialFields()) {
+            restartWarnings.add(Localization.lang("Serialize special fields"));
+        }
+
+        preferences.storeColumnPreferences(newColumnPreferences);
+    }
+
+    public String getFieldDisplayName(Field field) {
+        if (field instanceof SpecialField) {
+            return field.getName() + " (" + Localization.lang("Special") + ")";
+        } else if (field instanceof IEEEField) {
+            return field.getName() + " (" + Localization.lang("IEEE") + ")";
+        } else if (field instanceof InternalField) {
+            return field.getName() + " (" + Localization.lang("Internal") + ")";
+        } else if (field instanceof UnknownField) {
+            return field.getName() + " (" + Localization.lang("Custom") + ")";
+        } else if (field instanceof TableColumnsTabViewModel.ExtraFileField) {
+            return field.getName() + " (" + Localization.lang("File type") + ")";
+        } else {
+            return field.getName();
+        }
+    }
+
+    @Override
+    public boolean validateSettings() { return true; } // should contain at least one column
+
+    @Override
+    public List<String> getRestartWarnings() {
+        return restartWarnings;
+    }
+
+    public ListProperty<TableColumnsItemModel> columnsListProperty() { return this.columnsListProperty; }
+
+    public ObjectProperty<SelectionModel<TableColumnsItemModel>> selectedColumnModelProperty() { return selectedColumnModelProperty; }
+
+    public ListProperty<Field> availableColumnsProperty() { return this.availableColumnsProperty; }
+
+    public ObjectProperty<Field> addColumnProperty() { return this.addColumnProperty; }
+
+    public BooleanProperty showFileColumnProperty() { return this.showFileColumnProperty; }
+
+    public BooleanProperty showUrlColumnProperty() { return this.showUrlColumnProperty; }
+
+    public BooleanProperty preferUrlProperty() { return this.preferUrlProperty; }
+
+    public BooleanProperty preferDoiProperty() { return this.preferDoiProperty; }
+
+    public BooleanProperty specialFieldsEnabledProperty() { return this.specialFieldsEnabledProperty; }
+
+    public BooleanProperty specialFieldsSyncKeywordsProperty() { return this.specialFieldsSyncKeywordsProperty; }
+
+    public BooleanProperty specialFieldsSerializeProperty() { return this.specialFieldsSerializeProperty; }
+
+    public BooleanProperty showEPrintColumnProperty() { return this.showEPrintColumnProperty; }
+
+    public BooleanProperty extraFileColumnsEnabledProperty() { return this.extraFileColumnsEnabledProperty; }
+
+    public class ExtraFileField implements Field {
```

Comment:
I would actually move this class to a new file and put it next  to the other Fields.

Reply:
I am somewhat unsure about this. As this is just a wrapper class for the TableRowModel in the columnsList to draw the ExtraFileFieldColumns, I dont think it belongs into the model of Jabref. We already talked about removing these ExtraFileColumns completly, but after the discussion in #5244 i am very careful about removing any functionality...

Reply:
I created a util-class to store the ExtraFilePseudoField-class and a helper method which is going to be used also in PR #5265 . I hope this is a usable compromise.